### PR TITLE
refactor: add propagator stage

### PR DESCRIPTION
**Pull request by andiwand**
templated propagator stage for actors and aborters to disambiguate the callers state e.g. start of propagation, pre step, post step, end of propagation

### CODE REVIEW

**Comment by paulgessinger** [comment]
I'm wondering if at this point it would be cleaner to just use different methods?

**Comment by benjaminhuth** [comment]
I'm also not against specific functions, but so far it seems like it is rarely needed to know in which state we are... However, I think either way would be good

File: Core/include/Acts/Propagator/PropagatorStage.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+namespace Acts {
+
+enum class PropagatorStage {
+  start,
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/Propagator/PropagatorStage.hpp:14]: I agree, it's in line with `preStep` and `postStep`

**Review by asalzburger — APPROVED** [review]
From my point of view, with the renaming, we should start getting this development in, @andiwand can you address the comments?

**Comment by andiwand** [comment]
looks like I have a few more cases to handle after master went in again

**Comment by asalzburger** [comment]
@andiwand - there's a conflict, can you have a look ?

**Comment by andiwand** [comment]
I will close this for now as we do not have an urgent use for it and it looks like it complicates things

**Pull request closed without merging**